client/allocrunner/taskrunner/plugin_supervisor_hook.go

```diff
@@ -206,11 +195,29 @@ func (h *csiPluginSupervisorHook) Prestart(ctx context.Context,
 	return nil
 }
 
+func (h *csiPluginSupervisorHook) setSocketHook() {
+
+	// TODO(tgross): https://github.com/hashicorp/nomad/issues/11786
+	// If we're already registered, we should be able to update the
+	// definition in the update hook
+
+	// For backwards compatibility, ensure that we don't overwrite the
+	// socketPath on client restart with existing plugin allocations.
+	pluginInfo, _ := h.runner.dynamicRegistry.PluginForAlloc(
+		string(h.task.CSIPluginConfig.Type), h.task.CSIPluginConfig.ID, h.alloc.ID)
+	if pluginInfo != nil && pluginInfo.ConnectionInfo.SocketPath != "" {
+		h.socketPath = pluginInfo.ConnectionInfo.SocketPath
+	}
+
+	h.socketPath = filepath.Join(h.socketMountPoint, structs.CSISocketName)
```

Comment:
in the deleted code, this line is inside an `else`; should this be like that ?

Reply:
Oh shoot, good catch. I meant to add a `return` in the block above. Will fix.